src/tests/ftest/util/general_utils.py

```diff
@@ -90,7 +102,168 @@ def process_host_list(hoststr):
+    if log:
+        log.info("Connecting to the pool")
+    pool.connect(1 << 1)
+    return pool
+
+
+@fail_on(DaosApiError)
+def get_container(context, pool, log=None):
+    """Retrun a DAOS a container that has been created an opened.
+
+    Args:
+        context (DaosContext): the context to use to create the container
+        pool (DaosPool): pool in which to create the container
+        log (DaosLog|None): object for logging messages
+
+    Returns:
+        DaosContainer: an object representing a DAOS container
+
+    """
+    if log:
+        log.info("Creating a container")
+    container = DaosContainer(context)
+    container.create(pool.handle)
+    if log:
+        log.info("Opening a container")
+    container.open()
+    return container
+
+
+@fail_on(DaosApiError)
+def kill_server(server_group, context, rank, pool, log=None):
+    """Kill a specific server rank.
+
+    Args:
+        server_group (str):
+        context (DaosContext): the context to use to create the DaosServer
+        rank (int): daos server rank to kill
+        pool (DaosPool): the DaosPool from which to exclude the rank
+        log (DaosLog|None): object for logging messages
+
+    Returns:
+        None
+
+    """
+    if log:
+        log.info("Killing DAOS server {} (rank {})".format(server_group, rank))
+    server = DaosServer(context, server_group, rank)
+    server.kill(1)
+    if log:
+        log.info("Excluding server rank {}".format(rank))
+    pool.exclude([rank])
+
+
+@fail_on(DaosApiError)
+def get_pool_status(pool, log):
+    """Determine if the pool rebuild is complete.
+
+    Args:
+        pool (DaosPool): pool for which to determine if rebuild is complete
+        log (logging): logging object used to report the pool status
+
+    Returns:
+        None
+
+    """
+    pool_info = pool.pool_query()
+    message = "Pool: pi_ntargets={}".format(pool_info.pi_ntargets)
+    message += ", pi_nnodes={}".format(
+        pool_info.pi_nnodes)
+    message += ", pi_ndisabled={}".format(
+        pool_info.pi_ndisabled)
+    message += ", rs_version={}".format(
+        pool_info.pi_rebuild_st.rs_version)
+    message += ", rs_done={}".format(
+        pool_info.pi_rebuild_st.rs_done)
+    message += ", rs_toberb_obj_nr={}".format(
+        pool_info.pi_rebuild_st.rs_toberb_obj_nr)
+    message += ", rs_obj_nr={}".format(
+        pool_info.pi_rebuild_st.rs_obj_nr)
+    message += ", rs_rec_nr={}".format(
+        pool_info.pi_rebuild_st.rs_rec_nr)
+    message += ", rs_errno={}".format(
+        pool_info.pi_rebuild_st.rs_errno)
+    log.info(message)
+    return pool_info
+
+
+def is_pool_rebuild_complete(pool, log):
+    """Determine if the pool rebuild is complete.
+
+    Args:
+        pool (DaosPool): pool for which to determine if rebuild is complete
+        log (logging): logging object used to report the pool status
+
+    Returns:
+        bool: pool rebuild completion status
+
+    """
+    get_pool_status(pool, log)
```

Comment:
return type is pool_info need to put it in variable pool?

Reply:
Fixed.